dipy/workflows/vis_registration.py

```diff
@@ -0,0 +1,339 @@
+from __future__ import division, print_function, absolute_import
+from dipy.workflows.workflow import Workflow
+import logging
+import numpy as np
+import nibabel as nib
+from dipy.workflows.core import write_gif
+from dipy.io.image import load_affine_matrix
+from dipy.viz.regtools import overlay_slices
+from dipy.viz import window, actor
+
+from dipy.viz.window import snapshot
+
+
+class VisualizeRegisteredImage(Workflow):
+
+    def process_image_data(self, static_img, moved_img):
+
+        """
+        Function for pre-processing the image data. It involves
+        normalizing and copying static, moving images in the
+        red and green channel, respectively.
+
+        Parameters
+        ----------
+        static_img : array, shape (S, R, C) or (R, C)
+            The image to be used as reference during optimization.
+
+        moved_img : array, shape (S', R', C') or (R', C')
+            The image to be used as "moving" during optimization.
+
+        Returns
+        ------
+         overlay : ndarray
+            Containing the normalized image data
+
+         value_range : tuple
+            A tuple containing the minimum and maximum range of values to
+            select the pixels.
+        """
+
+        static_img = 255 * ((static_img - static_img.min()) /
+                            (static_img.max() - static_img.min()))
+        moved_img = 255 * ((moved_img - moved_img.min()) /
+                           (moved_img.max() - moved_img.min()))
+
+        # Create the color images
+        overlay = np.zeros(shape=static_img.shape + (3,), dtype=np.uint8)
+        overlay[..., 0] = static_img
+        overlay[..., 1] = moved_img
+        mean, std = overlay[overlay > 0].mean(), overlay[overlay > 0].std()
+        value_range = (mean - 0.5 * std, mean + 1.5 * std)
+        return overlay, value_range
+
+    def get_row_cols(self, num_slices):
+
+        """
+        Experimental helper function to get the number
+        of rows and columns for the mosaic.
+
+        num_slices: int
+            The number of slices as obtained from the
+            create_mosaic function.
+
+        Returns
+        -------
+        rows, cols : int, int
+            The number of rows and columns to be used in the grid for the
+            mosaic.
+        """
+        rows, cols = 0, 0
+
+        while True:
```

Comment:
compute it directly, no loops are needed
`num_slices += 5 - (num_slices % 5) if (num_slices % 5) > 0 else 0`